flink-yarn/src/main/java/org/apache/flink/yarn/YarnResourceManager.java

```diff
@@ -356,7 +363,8 @@ public void onContainersAllocated(List<Container> containers) {
 
 					final String containerIdStr = container.getId().toString();
 					final ResourceID resourceId = new ResourceID(containerIdStr);
-
+					this.resourceId = resourceId;
+					this.container = container;
```

Comment:
I think we can't do it like this, the `resourceId` will be reset by another request... as well as the `contianer`

Reply:
Yes. Thanks, I know what is wrong with my code.Will change.